### PR TITLE
Meta/print client

### PR DESCRIPTION
**Pull request by sorja**
image

Todo this pr:
- [x] Fix styling issues
- [x] Merge AssessmentPrintSection with AssessmentSection
- [x] ?

Todo next pr(s):
- [x]  OriginalDataPointsPrint
- [x] ContactPersonsPrintView

### CODE REVIEW

File: client/pages/AssessmentPrint/AssessmentPrint.tsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  const assessment = useAssessment()
+  const cycle = useCycle()
+  const countryIso = useCountryIso()
+  // useInitCountry()
+  // const assessmentType = useAssessmentType()
+  // const countryStatusLoaded = useIsCountryStatusLoaded()
+  const Component = Components[assessment?.props.name]
+
+  useEffect(() => {
+    dispatch(AssessmentActions.initApp())
```
Comment (minotogna) [inline client/pages/AssessmentPrint/AssessmentPrint.tsx:28]: Why is initApp needed here ?

File: client/pages/AssessmentPrint/FraPrint/useFraPrintDataFetch.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    dispatch(OriginalDataPointActions.fetchODPs({ countryIso }))
+  }, [])
+
+  const dataLoaded = useAppSelector(
+    (state) =>
+      tables.every((table) =>
+        AssessmentState.isSectionDataLoaded(table.assessmentType, table.sectionName, table.tableName)(state)
+      ) &&
+      // TODO add state objects
+      (state as any).userManagement.countryUsers &&
```
Comment (minotogna) [inline client/pages/AssessmentPrint/FraPrint/useFraPrintDataFetch.ts:41]: Use root state type

File: client/pages/AssessmentPrint/printAssessment.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import { getRequestParam } from '@webapp/utils/urlUtils'
+
+const printMode = 'print-mode'
+
+/**
+ * @deprecated
+ */
+export const setPrintingMode = () => document.body.classList.add(printMode)
+
+/**
```
Comment (minotogna) [inline client/pages/AssessmentPrint/printAssessment.ts:10]: Can we remove these deprecated consts?

File: client/pages/AssessmentSection/AssessmentSection.tsx [hunk 4]
```diff
@@ -1,39 +1,65 @@
...
 
   const panEuropean = assessmentName === AssessmentName.panEuropean
   const disabled = panEuropean || !canEditSection
 
-  if (!assessmentSection) return null
+  const shouldFetchTableSections = useRef(tableSections.length === 0)
+
+  useEffect(() => {
+    if (shouldFetchTableSections.current) {
+      shouldFetchTableSections.current = false
```
Comment (minotogna) [inline client/pages/AssessmentSection/AssessmentSection.tsx:44]: Why fetching here ?

File: client/pages/AssessmentSection/Descriptions/Descriptions.tsx [hunk 5]
```diff
@@ -45,7 +46,7 @@ const Descriptions: React.FC<Props> = (props: Props) => {
   const { descriptions, disabled, sectionName } = props
 
   const i18n = useTranslation()
-  const [printView, printOnlyTablesView] = [false, false] // TODO: usePrintView()
+  const { print, onlyTables } = useIsPrint()
 
   const { introductoryText } = descriptions
   const { analysisAndProcessing, nationalData } = useDescriptions(props)
```
Comment (minotogna) [inline client/pages/AssessmentSection/Descriptions/Descriptions.tsx:52]: Check useDeacriptions hook , it should check print as well

File: client/pages/AssessmentPrint/AssessmentPrint.tsx (hunk 1 above)
Comment (sorja) [reply client/pages/AssessmentPrint/AssessmentPrint.tsx:28]: PrintView has no AppInit call (opens in separate window/tab)

File: client/pages/AssessmentSection/AssessmentSection.tsx (hunk 4 above)
Comment (sorja) [reply client/pages/AssessmentSection/AssessmentSection.tsx:44]: We have section info only here (todo: pass array of sections for endpoint)

File: client/pages/AssessmentPrint/FraPrint/useFraPrintDataFetch.ts (hunk 2 above)
Comment (sorja) [reply client/pages/AssessmentPrint/FraPrint/useFraPrintDataFetch.ts:41]: Deprecated file will be removed in PR:  OriginalDataPointsPrint

File: client/pages/AssessmentPrint/AssessmentPrint.tsx (hunk 1 above)
Comment (sorja) [reply client/pages/AssessmentPrint/AssessmentPrint.tsx:28]: Removed - only sections need to be fetched

File: client/pages/AssessmentPrint/FraPrint/useFraPrintDataFetch.ts [hunk 6]
```diff
@@ -0,0 +1,49 @@
+import { useEffect } from 'react'
+import { useDispatch } from 'react-redux'
+
+import { FRA } from '@core/assessment'
+import { Objects } from '@core/utils'
+import { batchActions, useAppSelector } from '@webapp/store'
+import { SectionSpecs, TableSummarySpec } from '@webapp/sectionSpec'
+import * as AssessmentState from '@webapp/app/assessment/assessmentState'
+import { fetchTableData } from '@webapp/components/Assessment/DataTable/actions'
+import { fetchUsers } from '@webapp/app/user/userManagement/actions'
```
Comment (minotogna) [inline client/pages/AssessmentPrint/FraPrint/useFraPrintDataFetch.ts:10]: Use convention to import actions

File: client/pages/AssessmentSection/AssessmentSection.tsx (hunk 4 above)
Comment (minotogna) [reply client/pages/AssessmentSection/AssessmentSection.tsx:44]: Don't we do that in the section wrapper ?
We should do it only once

File: client/pages/AssessmentSection/Title/ExtentOfForest/ExtentOfForest.tsx [hunk 7]
```diff
@@ -19,28 +21,27 @@ const ExtentOfForest: React.FC<Props> = (props) => {
...
       <h2 className="headline no-print">
         {i18n.t<string>(`${sectionName}.${sectionName}`)}
         {Users.isAdministrator(user) && hasOdps && (
-          <button
-            type="button"
-            className="btn-s btn-secondary"
-            style={{ marginLeft: '12px' }}
-            onClick={() => dispatch(AssessmentSectionActions.toggleShowOriginalDataPoint())}
-          >
+          <button type="button" className="btn-s btn-secondary" style={{ marginLeft: '12px' }} onClick={onClick}>
```
Comment (rfontanarosa) [inline client/pages/AssessmentSection/Title/ExtentOfForest/ExtentOfForest.tsx:38]: move style to an external file

File: client/pages/AssessmentPrint/FraPrint/ContactPersons/ContactPersons.tsx [hunk 8]
```diff
@@ -0,0 +1,64 @@
+import './ContactPersons.scss'
```
Comment (rfontanarosa) [inline client/pages/AssessmentPrint/FraPrint/ContactPersons/ContactPersons.tsx:1]: maybe should be renamed into ContactPeople

Comment (minotogna) [reply client/pages/AssessmentPrint/FraPrint/ContactPersons/ContactPersons.tsx:1]: mmm or just Contacts if you really want to rename it

Comment (sorja) [reply client/pages/AssessmentPrint/FraPrint/ContactPersons/ContactPersons.tsx:1]: I would leave as is, if you really want to rename, let me know!

**Review by minotogna — APPROVED** [review]

**Review by rfontanarosa — APPROVED** [review]

**Pull request merged**